src/codegen/compile.ml

```diff
@@ -3894,7 +3894,11 @@ module IC = struct
     let fi = E.add_fun env "canister_heartbeat"
       (Func.of_body env [] [] (fun env ->
         G.i (Call (nr (E.built_in env "heartbeat_exp"))) ^^
-        GC.collect_garbage env))
+        (* TODO(3622)
```

Comment:
I saw this while implementing `system func timer`, and I remember considering this a bit expensive/dangerous. But with DTS not covering this, yuck!

Reply:
Yeah, I remember you suggesting it but thought better no to mess with the GC. This just a temp hack. Let's see what Luc thinks.